librz/include/rz_analysis.h

```diff
@@ -999,8 +999,8 @@ typedef struct rz_analysis_esil_trace_t {
 	ut64 stack_addr;
 	ut64 stack_size;
 	ut8 *stack_data;
-	//TODO remove `db` and reuse info above
-	Sdb *db;
+	// RzVector<RzILTraceInstruction>
+	RzPVector *instructions;
```

Comment:
There was an interesting comment done by @pelijah about the amount of memory used by this trace system... I think that comment was right indeed, though I also believe we do not use much memory (if more at all) than before, given that everything was a string.

I believe optimizations should be left for later, but something you can do immediately is probably to use `RzPVector instructions` instead of `RzPVector *instructions`. Maybe also in `RzILTraceInstruction` you can do the same? However, as said, I'm a bit worried of this optimization without having data about it... @thestr4ng3r @pelijah ?

Reply:
I agree about saving optimizations for later.